go/chat/uithreadloader.go

```diff
@@ -76,11 +77,25 @@ func (t *UIThreadLoader) groupThreadView(ctx context.Context, uid gregor1.UID, t
 
 	// group JOIN/LEAVE messages
 	newMsgs := t.groupGeneric(ctx, uid, tv.Messages,
-		func(msg chat1.MessageUnboxed) bool {
+		func(msg chat1.MessageUnboxed, grouped []chat1.MessageUnboxed) bool {
 			body := msg.Valid().MessageBody
 			mtyp, err := body.MessageType()
-			return (err == nil && (mtyp == chat1.MessageType_JOIN ||
-				mtyp == chat1.MessageType_LEAVE) && !msg.Valid().ClientHeader.Sender.Eq(uid))
+			if err != nil {
+				return false
+			}
+			if !(mtyp == chat1.MessageType_JOIN || mtyp == chat1.MessageType_LEAVE) {
+				return false
+			}
+			if msg.Valid().ClientHeader.Sender.Eq(uid) {
+				return false
+			}
+			for _, g := range grouped {
```

Comment:
might be better to have a global join/leave map so we don't have to traverse the loop repeatedly

Reply:
This is the current set of consolidated messages. It won't be long.